### PR TITLE
[Topology] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ VoxelGridLoader::VoxelGridLoader()
       roi ( initData ( &roi, Vec6i ( 0,0,0, 0xFFFF, 0xFFFF, 0xFFFF ), "ROI", "Region of interest (xmin, ymin, zmin, xmax, ymax, zmax)" ) ),
       headerSize ( initData ( &headerSize, 0, "header", "Header size in bytes" ) ),
       segmentationHeaderSize ( initData ( &segmentationHeaderSize, 0, "segmentationHeader", "Header size in bytes" ) ),
-      idxInRegularGrid(initData(&idxInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
+      idxInRegularGrid(initData(&idxInRegularGrid,"d_idxInRegularGrid","indices of the hexa in the grid.")),
```
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp]:
```suggestion
      idxInRegularGrid(initData(&idxInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
```
and you did not renamed the Data...

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/DynamicSparseGridTopologyContainer.cpp [hunk 2]
```diff
@@ -40,16 +40,25 @@ int DynamicSparseGridTopologyContainerClass = core::RegisterObject ( "Hexahedron
...
+    , d_resolution (initData (&d_resolution, type::Vec3i (0, 0, 0 ), "resolution", "voxel grid resolution" ) )
+    , d_valuesIndexedInRegularGrid(initData (&d_valuesIndexedInRegularGrid, sofa::type::vector<unsigned char>(), "d_valuesIndexedInRegularGrid", "values indexed in the Regular Grid" ) )
+    , d_valuesIndexedInTopology(initData(&d_valuesIndexedInTopology, "d_valuesIndexedInTopology", "values indexed in the topology"))
+    /// deprecated
+    ,valuesIndexedInTopology(initData(&valuesIndexedInTopology, "valuesIndexedInTopology", "values indexed in the topology"))
+    , d_idxInRegularGrid(initData (&d_idxInRegularGrid, sofa::type::vector<BaseMeshTopology::HexaID>(), "d_idxInRegularGrid", "indices in the Regular Grid" ) )
+    , d_idInRegularGrid2IndexInTopo(initData (&d_idInRegularGrid2IndexInTopo, std::map< unsigned int, BaseMeshTopology::HexaID> (), "d_idInRegularGrid2IndexInTopo", "map between id in the Regular Grid and index in the topology" ) )
+    , d_voxelSize(initData(&d_voxelSize, type::Vec3(1_sreal, 1_sreal, 1_sreal), "voxelSize", "Size of the Voxels"))
 {
```
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/DynamicSparseGridTopologyContainer.cpp:51]: Please, review all data factory name. e.g. : `    , d_valuesIndexedInRegularGrid(initData (&d_valuesIndexedInRegularGrid, sofa::type::vector<unsigned char>(), "d_valuesIndexedInRegularGrid", "values indexed in the Regular Grid" ) )` --> the third parameter **should not be modified**

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/MultilevelHexahedronSetTopologyContainer.cpp [hunk 3]
```diff
@@ -45,7 +45,7 @@ MultilevelHexahedronSetTopologyContainer::MultilevelHexahedronSetTopologyContain
     : HexahedronSetTopologyContainer(),
       _level(initData(&_level, 0, "level", "Number of resolution levels between the fine and coarse mesh")),
       fineResolution(initData(&fineResolution,Vec3i(0,0,0),"resolution","fine resolution")),
-      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
+      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"d_idxInRegularGrid","indices of the hexa in the grid.")),
```
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/MultilevelHexahedronSetTopologyContainer.cpp]:
```suggestion
      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
```

File: Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/SubsetTopologicalMapping.cpp [hunk 4]
```diff
@@ -59,25 +59,43 @@ int SubsetTopologicalMappingClass = core::RegisterObject("This class is a specif
...
+    , d_edgeD2S(initData(&d_edgeD2S, "d_edgeD2S", "Internal destination -> source topology edges map"))
+    , d_triangleS2D(initData(&d_triangleS2D, "d_triangleS2D", "Internal source -> destination topology triangles map"))
+    , d_triangleD2S(initData(&d_triangleD2S, "d_triangleD2S", "Internal destination -> source topology triangles map"))
+    , d_quadS2D(initData(&d_quadS2D, "d_quadS2D", "Internal source -> destination topology quads map"))
+    , d_quadD2S(initData(&d_quadD2S, "d_quadD2S", "Internal destination -> source topology quads map"))
+    , d_tetrahedronS2D(initData(&d_tetrahedronS2D, "d_tetrahedronS2D", "Internal source -> destination topology tetrahedra map"))
+    , d_tetrahedronD2S(initData(&d_tetrahedronD2S, "d_tetrahedronD2S", "Internal destination -> source topology tetrahedra map"))
+    , d_hexahedronS2D(initData(&d_hexahedronS2D, "d_hexahedronS2D", "Internal source -> destination topology hexahedra map"))
+    , d_hexahedronD2S(initData(&d_hexahedronD2S, "d_hexahedronD2S", "Internal destination -> source topology hexahedra map"))
 {
```
Comment (bakpaul) [inline Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/SubsetTopologicalMapping.cpp:80]: Please here also, remove all modification of factory data name.

**Review by bakpaul — CHANGES_REQUESTED** [review]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
Note that there are still Data variables that are not been renamed in `Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.h`

**Review by bakpaul — APPROVED** [review]

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
> Note that there are still Data variables that are not been renamed in `Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.h`

You intend to rename them in this PR, or should we merge as it is ?

**Pull request merged**